config/config.go

```diff
@@ -101,3 +101,11 @@ func (c *Config) ProcessOptions(options ...Option) {
 		}
 	}
 }
+
+// IsValidName validates configuration objects supported by libnetwork
+func IsValidName(name string) bool {
```

Comment:
Agreed. actually am using this for network, endpoint and drivers now. All of these have the same requirement. We can enhance it to a regex as we see more requirements for any such restrictions.